src/foam/u2/crunch/wizardflow/RequirementsPreviewAgent.js

```diff
@@ -27,9 +27,19 @@ foam.CLASS({
     // If Property expressions ever unwrap promises this method can be blank.
     function execute() {
       var sectionsList = this.generateSections(this.wizardlets);
-      var arrOfRequiredCapabilities = sectionsList.flat()
-        .filter(eachSection => eachSection && eachSection.help)
-        .map(eachSection => eachSection.help);
+      const arrOfRequiredCapabilities = [];
+      for ( let i = 0; i < sectionsList.length; i++ ) {
+        for ( let j = 0; j < sectionsList[i].length; j++ ) {
+          if ( sectionsList[i][j].help ) {
+            if ( typeof sectionsList[i][j].help === 'function' ) {
+              let curCap = this.wizardlets[i].capability;
+              let generateHelpText = sectionsList[i][j].help.bind(curCap.of.create({}, this));
+              sectionsList[i][j].help = generateHelpText(curCap);
+            }
+            arrOfRequiredCapabilities.push(sectionsList[i][j].help);
+          }
+        }
+      }
```

Comment:
If you want to add this logic into the previous `.map` function using ternary operators that would make the code smaller and easier to read. That's just a suggestion though, this has my approval.

Reply:
^ After discussion with @jinn9 the functional approach would be necessarily messy here because the unflattened sections are needed, so my suggestion has been voided